csharp/test/Drivers/Interop/Snowflake/CastTests.cs

```diff
@@ -0,0 +1,397 @@
+        [InlineData("OBJECT", "SELECT OBJECT_CONSTRUCT('fortyTwo', 42::NUMERIC)", "{\n  \"fortyTwo\": 42\n}", ArrowTypeId.String, "TO_VARIANT")]
+        [InlineData("VARCHAR", "SELECT 42", "\"42\"", ArrowTypeId.String, "TO_VARIANT")]
+        [InlineData("VARIANT", "SELECT 42::VARIANT", "42", ArrowTypeId.String, "TO_VARIANT")]
+        [InlineData("VARIANT", "SELECT 'JONES'::VARIANT", "\"JONES\"", ArrowTypeId.String, "TO_VARIANT")]
+        public void TestCastPositiveStructured(string columnSpecification, string sourceValue, object expectedValue, ArrowTypeId expectedType, string castFunction, string castExpression = null)
+        {
+            InitializeTest(columnSpecification, sourceValue, out string columnName, out string table, useSelectSyntax: true);
+            SelectWithCastAndValidateValue(
+                table,
+                castFunction,
+                castExpression ?? columnName,
+                expectedValue,
+                expectedType);
+        }
+
+        [SkippableTheory]
+        [InlineData("VARCHAR", "'ABC'", ArrowTypeId.Decimal128, "TRY_TO_NUMERIC", null)] // Non numeric value
+        [InlineData("VARCHAR", "'3'", ArrowTypeId.Boolean, "TRY_TO_BOOLEAN", null)] // Non boolean value
+        [InlineData("VARCHAR", "'31/12/1970'", ArrowTypeId.Timestamp, "TRY_TO_TIMESTAMP_TZ", null)] // Non date value
+        public void TestTryCastNegative(
+            string columnSpecification,
+            string sourceValue,
+            ArrowTypeId expectedType,
+            string castFunction,
+            string castExpression = null)
+        {
+            {
+                InitializeTest(columnSpecification, sourceValue, out string columnName, out string table);
+                SelectWithCastAndValidateValue(
+                    table,
+                    castFunction,
+                    castExpression ?? columnName,
+                    null, // Returns null if value cannot be converted.
+                    expectedType);
+            }
+        }
+
+        private void InitializeTest(string columnSpecification, string sourceValue, out string columnName, out string table, bool useSelectSyntax = false)
+        {
+            columnName = COLUMN_NAME;
+            table = CreateTemporaryTable(
+                _statement,
+                s_testTablePrefix,
+                _catalogSchema,
+                string.Format("{0} {1}", columnName, columnSpecification));
+            if (useSelectSyntax)
+            {
+                InsertIntoFromSelect(table, columnName, sourceValue);
+            }
+            else
+            {
+                InsertSingleValue(table, columnName, sourceValue);
+            }
+        }
+
+        private void InsertSingleValue(string table, string columnName, string value)
+        {
+            string insertNumberStatement = string.Format("INSERT INTO {0} ({1}) VALUES ({2});", table, columnName, value);
+            Console.WriteLine(insertNumberStatement);
```

Comment:
Do we want to retain these console writes? Is there a logger in the framework that can be used instead?

Reply:
@jduo - Do you want me to replace the `Console` usage in all tests?

Reply:
Done.

Reply:
Opened #1468 to remove Console.WriteLine.